powersimdata/data_access/data_access.py

```diff
@@ -27,6 +31,27 @@ def __init__(self, root):
         self.root = root
         self.join = fs2.path.join
 
+    def write(self, filepath, data):
+        """Write a file from data store,
+
+        :param str filepath: path to save data to, with extension either 'pkl', 'csv', or 'mat'.
+        :param (*pandas.DataFrame* or *dict*) data: data to save
+        :raises ValueError: if extension is unknown.
+        """
+
+        ext = os.path.basename(filepath).split(".")[-1]
+        if ext == "pkl":
+            self.fs.writebytes(filepath, pickle.dumps(data))
+        elif ext == "csv":
+            self.fs.writetext(filepath, data.to_csv())
+        elif ext == "mat":
+            tmp = io.BytesIO()
+            savemat(tmp, data, appendmat=False)
+            tmp.seek(0)
+            self.fs.writefile(filepath, tmp)
```

Comment:
I think we could standardize a bit. What I mean is, each of `pickle.dump`, `data.to_csv` and `savemat` can take a file object to write to. So if we did `with self.fs.openbin(filepath) as f:` before the if/else, then inside could just be `savemat(f, data, appendmat=False)`, etc. Of course, this only works if those calls don't assume it's an os file, and stick with the standard file api. So I see the advantage to the current approach, just thought we might try eliminating the extra write to the temp file here.

Reply:
Thanks! I didn't see the `openbin` option when looking through pyfilesystem, so thanks for pointing that out!

Reply:
This looks good. Did you run through usage examples again? Just want to make sure the 3 methods don't have any hidden assumption about the file object being specifically an os file.